lib/src/Base/Type/SymmetricMatrix.cxx

```diff
@@ -325,5 +326,13 @@ Scalar SymmetricMatrix::computeSumElements() const
   return getImplementation()->computeSumElements();
 }
 
+/** Compute inverse */
+SymmetricMatrix SymmetricMatrix::inverse() const
```

Comment:
It should be done at the MatrixImplementation level. You should add the following methods:
 * MatrixImplementation::inverseSquare() const
 * MatrixImplementation::inverseSquareInPlace()
 * MatrixImplementation::inverseSymmetric() const
 * MatrixImplementation::inverseSymmetricInPlace() const

then use these methods to implement the SquareMatrix::inverse() and SymmetricMatrix::inverse() methods, the same way the SquareMatrix::solveLinearSystem()/MatrixImplementation::solveLinearSystemSquare() methods are implemented.

Reply:
I do not really understand why this is the right way to go, but I followed that and fixed it at 447c802e8.

Reply:
Because the pattern interface/implementation is specific for `Matrix`/`MatrixImplementation`
Usually classes inherit from the `implementation`, but here it inherit from the `interface` one.
Thus we need to write the method at the top level and redirect in the new classes (`SquareMatrix`, `SymmetricMatrtix`) to use the specific method

Reply:
OK, I did not notice that. So why does the implementations inherit from the implementation in this particular case, and not in other classes? What is specific to matrices?

Reply:
Ok. Fixed at 447c802e8a3d8c6688a1a1b1b5b1c80e30ec42e5